src/main/scala/workflow/GatherTransformer.scala

```diff
@@ -0,0 +1,15 @@
+package workflow
+
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+private[workflow] class GatherTransformer[T] extends TransformerNode[Seq[T]] {
+  def transform(dataDependencies: Seq[_], fitDependencies: Seq[TransformerNode[_]]): Seq[T] = dataDependencies.map(_.asInstanceOf[T])
+
+  def transformRDD(dataDependencies: Seq[RDD[_]], fitDependencies: Seq[TransformerNode[_]]): RDD[Seq[T]] = {
+    dataDependencies.map(_.asInstanceOf[RDD[T]].map(t => Seq(t))).reduceLeft((x, y) => {
+      x.zip(y).map(z => z._1 ++ z._2)
```

Comment:
@tomerk and I discussed this yesterday. We used to have a Zip `FunctionNode` that did this, and agreed that it might be possible to make the DAG smaller. Should we add an issue? In general I think we're looking at n < 100 for most of our uses.